phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/utils/CsvWriter.kt

```diff
@@ -0,0 +1,149 @@
+package fr.acinq.phoenix.utils
+
+import fr.acinq.phoenix.data.WalletPaymentInfo
+import fr.acinq.phoenix.utils.extensions.createdAt
+import kotlinx.datetime.Instant
+
+class CsvWriter {
+    
+    data class Configuration(
+        val includesFiat: Boolean,
+        val includesDescription: Boolean,
+        val includesNotes: Boolean
+    )
+
+    companion object {
+        private const val FIELD_ID          = "ID"
+        private const val FIELD_DATE        = "Date"
+        private const val FIELD_AMOUNT_BTC  = "Amount BTC"
+        private const val FIELD_AMOUNT_FIAT = "Amount Fiat"
+        private const val FIELD_DESCRIPTION = "Description"
+        private const val FIELD_NOTES       = "Notes"
+
+        /**
+         * Creates and returns the header row for the CSV file.
+         * This includes the CRLF that terminates the row.
+         */
+        fun makeHeaderRow(config: Configuration): String {
+            var header = "$FIELD_ID,$FIELD_DATE,$FIELD_AMOUNT_BTC"
+            if (config.includesFiat) {
+                header += ",$FIELD_AMOUNT_FIAT"
+            }
+            if (config.includesDescription) {
+                header += ",$FIELD_DESCRIPTION"
+            }
+            if (config.includesNotes) {
+                header += ",$FIELD_NOTES"
+            }
+
+            header += "\r\n"
+            return header
+        }
+
+        /**
+         * Creates and returns the row for the given payment.
+         * This includes the CRLF that terminates the row.
+         *
+         * @param info Payment fetched via PaymentsFetcher
+         * @param localizedDescription As displayed in app (might be a localized default value)
+         * @param config The configuration for the CSV file
+         */
+        fun makeRow(
+            info: WalletPaymentInfo,
+            localizedDescription: String,
+            config: Configuration
+        ): String {
+
+            val id = info.id().dbId
+            var row = processField(id)
+
+            val completedAt = info.payment.completedAt()
+            val date = if (completedAt > 0) completedAt else info.payment.createdAt
+            val dateStr = Instant.fromEpochMilliseconds(date).toString() // ISO-8601 format
+            row += ",${processField(dateStr)}"
+
+            val amtBtc = "${info.payment.amount.msat} msat"
+            row += ",${processField(amtBtc)}"
```

Comment:
I think we should not append `msat`, instead the bitcoin amount column name should contain the unit, but the values should be 100% numeric. That way the user can easily order or format this column from his CSV editor.

Reply:
That's a good idea